specifyweb/frontend/js_src/lib/components/FormSliders/RecordSet.tsx

```diff
@@ -412,7 +416,33 @@ function RecordSet<SCHEMA extends AnySchema>({
                 })
             : undefined
         }
-        onClone={(newResource): void => go(totalCount, 'new', newResource)}
+        onClone={(resources: RA<SpecifyResource<SCHEMA>>): void => {
+          go(totalCount, 'new', resources[0]);
+          if (resources.length > 1) {
+            const sortedResources = Array.from(resources).sort(
+              sortFunction((r) => r.id)
+            );
+            loading(
+              createNewRecordSet(
+                sortedResources.map((resource) => resource.id)
+              ).then(async () => {
+                const startingResourceCatNumber =
+                  serializeResource(
+                    sortedResources[0] as SpecifyResource<CollectionObject>
+                  ).catalogNumber ?? '0';
+                const endingResourceCatNumber =
+                  serializeResource(
+                    sortedResources.at(-1) as SpecifyResource<CollectionObject>
+                  ).catalogNumber ?? '0';
+                recordSet.set(
+                  'name',
+                  `Batch #${startingResourceCatNumber} - #${endingResourceCatNumber}`
+                );
+                await recordSet.save();
+              })
+            );
+          }
```

Comment:
We only want to execute this if the table is `CollectionObject`, no? 
This code could conceptually be from any table, which is why you received the errors mentioned in
https://github.com/specify/specify7/pull/4804/files#r1649414105. 

Make sure this code is _only_ run if recources.length > 1 and the table is `CollectionObject`.
Then you can proceed with the suggestions in https://github.com/specify/specify7/pull/4804/files#r1649414105 (no need to serialize the resources, just use `resource.get('catalogNumber')`). 

If typescript is not smart enough to infer that `sortedResources` is of type `RA<SpecifyResource<CollectionObject>>`, then we can do type casting because we _know_ they types will always be correct within the if statement.